docs/cromwell_features/HogFactors.md

```diff
@@ -0,0 +1,163 @@
+- Within that limit, hog factors now allow us to limit the maximum concurrent jobs started per hog group.
+    + This means that new workflows can immediately run even if big workflows are queued up.
+
+#### Round robin allocation
+
+Rather than assigning job tokens on a strict first-come first served basis, Cromwell now assigns in a round-robin
+fashion between hog groups and then on a first-come-first-served *within* a hog group.
+
+In other words if the hog groups had the following entries queued up:
+```
+ A: jobA1, jobA2, jobA3, ..., jobA1000000
+ B: jobB1, jobB2
+ C: jobC1
+ D: jobD1, jobD2
+```
+
+Then Cromwell would start the jobs in the following order, even though `jobA1000000` was added before `jobD1`:
+```
+jobA1, jobB1, jobC1, jobD1, jobA2, jobB2, jobD2, jobA3, ..., jobA1000000
+```
+
+#### Example: How job execution is affected by hog factors
+
+##### An administrator sets up a Cromwell server
+
+- A Cromwell administrator sets the overall maximum concurrent job limit to 100,000 PAPIv2 jobs.
+- The administrator also sets the hog factor to be 25.
+- Cromwell will therefore calculate a per-hog-group concurrent job limit of 4,000 PAPIv2 jobs.
+
+##### Our first hog group hits its limit
+
+- 100 workflows are running in hog group "A" and between them want to send 20,000 jobs to PAPIv2. 
+    + Cromwell will initially limit that to 4,000 and only start new jobs when existing jobs from this group finish.
+    + New workflows in this group will not be able to start jobs either until existing jobs complete.
+    + Note that Cromwell is only using 1/25th of its limit even though it would otherwise be able to go faster.
+
+##### Another hog group appears
+
+- Now hog group B submits 1,000 workflows and between them they want to send 200,000 jobs to PAPIv2.
+- Even though group A's jobs are only trickling in, group B's workflows are allowed to start 4,000 jobs immediately,
+- The rest of group B's jobs are queued up waiting for group B's existing jobs to complete.
+
+##### Where do we stand?
+
+- Cromwell knows about 220,000 jobs that could be started
+- Cromwell has an overall limit of 100,000
+- Cromwell is running 8,000 jobs in two hog groups.
+
+In other words, not so great - perhaps we should have set the hog factor lower...?
+
+#####But wait, more workflows appear...
+
+- Now another 23 hog groups ("C" through "Y") submit workflows of a similar scale to hog group A.
+- One by one, the workflows of each hog group fill up their share of the overall concurrent job limit.
+- So Cromwell is now running 100,000 jobs and each hog group has been allocated 4,000 of those.
+
+##### What about poor hog group "Z"?
+
+- A final group submits workflows under hog group "Z".
+- Alas, even though hog group "Z" is not running anything yet, we cannot start their workflows because we're 
+at the global maximum of 100,000.
+
+
+In other words, perhaps we should have set the hog factor higher...?
```

Comment:
Again, might want to italicize or otherwise indicate that this line is a note of reflection